Doc/library/pickle.rst

```diff
@@ -629,6 +629,12 @@ or both.
      value``.  This is primarily used for dictionary subclasses, but may be used
      by other classes as long as they implement :meth:`__setitem__`.
 
+   * Optionally, a callable with a ``(obj, state)`` signature. This
+     callable allows the user to programatically control the state-updating
+     behavior of a specific object, instead of using ``obj``'s static
+     :meth:`__setstate__` method. If not ``None``, this callable will have
+     priority over ``obj``'s :meth:`__setstate__`.
+
```

Comment:
Please add a marker :

```
    .. versionadded:: 3.8
```

I am not sure about the required indentation level for the versionadded directive.